src/main/java/pl/sda/library/infrastructure/jpa/BookDtoCreator.java

```diff
@@ -32,19 +32,34 @@ BookDto fromDomain(Book book) {
 
     private SeriesDto getSeriesFromString(String series) {
         Optional<SeriesDto> seriesDtoByTitle = seriesDtoRepository.findSeriesDtoByTitle(series);
-        if (seriesDtoByTitle.isPresent())
-            return seriesDtoByTitle.get();
-        return null;
+        if (seriesDtoByTitle.isPresent()){
+            return seriesDtoByTitle.get();}
+        else {
+            SeriesDto seriesDto = new SeriesDto();
+            seriesDto.setTitle(series);
+            seriesDto.setDescription(series);
+            seriesDto.setId(seriesDtoRepository.save(seriesDto).getId());
+            return seriesDto;
+        }
     }
 
+
     private Set<AuthorDto> getAuthorsFromString(String authors) {
         Set<AuthorDto> authorDtos = new HashSet<>();
         String[] authorsList = authors.trim().split(",");
         for (String author : authorsList) {
             String[] authorsSplit = author.trim().split(" ");
             Optional<AuthorDto> authorDtoByFirstNameAndLastName = authorDtoRepository.findAuthorDtoByFirstNameAndLastName(authorsSplit[0], authorsSplit[1]);
-            if (authorDtoByFirstNameAndLastName.isPresent())
+            if (authorDtoByFirstNameAndLastName.isPresent()) {
                 authorDtos.add(authorDtoByFirstNameAndLastName.get());
+            }
+            else {
+                AuthorDto newAuthorDto = new AuthorDto();
+                newAuthorDto.setFirstName(authorsSplit[0]);
+                newAuthorDto.setLastName(authorsSplit[1]);
+                newAuthorDto.setId(authorDtoRepository.save(newAuthorDto).getId());
+                authorDtos.add(newAuthorDto);
```

Comment:
odwołujesz się bezpośrednio do save czyli jak nie będzie to doda a będzie to nadpisze
mamy add które zwraca id
newAuthorDto.setId(authorDtoRepository.add(newAuthorDto));

albo jak chcesz z save
 authorDtos.add(authorDtoRepository.save(newAuthorDto).get());